реализована регистрация, авторизация, сброс пароля и обновление токена через Server Actions

kairat-genius opened the pull request:
- Добавлены Server Actions для сохранения, обновления и удаления токенов
- Реализована логика сброса пароля с поддержкой параметров в URL
- Добавлены проверки наличия access и refresh токенов для ограничения доступа к /profile
- Обновлён компонент направления (Direction) на главной странице
- Добавлена глобальная проверка авторизованности пользователя 
- Добавлена кастомная toast-уведомление

@web-dev-max

kairat-genius committed:
реализована регистрация, авторизация, сброс пароля и обновление токена через Server Actions

kairat-genius committed:
удалена ненужная функция

kairat-genius committed:
перенесены get user и обновление refresh токена с клиента на сервер

web-dev-max commented on new-frontend/src/features/modal/login-modal/ui/LoginModal.tsx:
```diff
@@ -1,31 +1,73 @@
-  const onSubmit: SubmitHandler<LoginUserType> = async data => {
-    console.log(data)
-  }
+  const onSubmit: SubmitHandler<LoginType> = async data => {
```
Блок try catch. Да и в целом запросы лучше хранить в сторе или где то в отдельном файле связанный с api. А в компоненте просто делать вызов

web-dev-max commented on new-frontend/src/features/modal/register-modal/ui/RegisterModal.tsx:
```diff
@@ -1,44 +1,81 @@
-    console.log(data)
-  }
+    setHasManualEmailError(false);
+    try {
```
Блок try catch. Да и в целом запросы лучше хранить в сторе или где то в отдельном файле связанный с api. А в компоненте просто делать вызов

web-dev-max commented on new-frontend/src/features/modal/reset-password-modal/ui/ResetPasswordForm.tsx:
```diff
@@ -1,27 +1,51 @@
   const onSubmit: SubmitHandler<ResetPasswordFormType> = async data => {
-    console.log(data)
+
+    try {
```
Блок try catch. Да и в целом запросы лучше хранить в сторе или где то в отдельном файле связанный с api. А в компоненте просто делать вызов

web-dev-max commented on new-frontend/src/features/modal/reset-password-modal/ui/ResetPasswordRequest.tsx:
```diff
@@ -1,31 +1,61 @@
-    console.log(data)
-  }
+    setHasManualEmailError(false);
+    try {
```
Блок try catch. Да и в целом запросы лучше хранить в сторе или где то в отдельном файле связанный с api. А в компоненте просто делать вызов

web-dev-max commented on new-frontend/src/features/profile/logout/index.tsx:
```diff
@@ -1,10 +1,21 @@
+  const handleLogout = async () => {
+    await removeTokens()
+    startTransition(() => {
+      router.push('/')
```
Перед вызово роутера, лучше добавить проверку, точно ли удалился токен. Иначе можно схватить баг

kairat-genius committed:
feat: подключены статичные данные, улучшена структура и безопасность проекта

- Подключены и использованы статичные данные с backend
- Перенесена логика try-catch из компонентов и форм в сервисные функции
- Разбит eader на подкомпоненты для улучшения читаемости
- Добавлен isomorphic-dompurify для защиты dangerouslyetnner от -атак
- Динамически загружаемые компоненты переведены на next/dynamic с uspense и fallback

kairat-genius commented:
подключены статичные данные, улучшена структура и безопасность проекта

- Подключены и использованы статичные данные с backend API
- Перенесена логика try-catch из компонентов и форм в сервисные функции
- Разбит Header на переиспользуемые подкомпоненты для улучшения читаемости
- Добавлен isomorphic-dompurify для защиты dangerouslySetInnerHTML от XSS-атак
- Динамически загружаемые компоненты переведены на next/dynamic с Suspense и fallback
@web-dev-max
